include/SDL3/SDL_intrin.h

```diff
@@ -101,21 +101,23 @@ _m_prefetch(void *__P)
 #include <lasxintrin.h>
 #define __LASX__
 #endif
-#if defined(HAVE_IMMINTRIN_H) && !defined(SDL_DISABLE_IMMINTRIN_H)
+#if defined(__AVX__) && !defined(SDL_DISABLE_AVX)
 #include <immintrin.h>
-#else
-#if defined(__MMX__) && !defined(SDL_DISABLE_MMINTRIN_H)
+#endif
+#if defined(__MMX__) && !defined(SDL_DISABLE_MMX)
```

Comment:
I didn't realize that this was a public header. I think this is still fine, but keep in mind that the cmake detection flags aren't in any public headers, so third party projects have to define these themselves if they want to avoid including the intrinsics headers. We intentionally don't include this header in SDL.h and just provide it as a convenience for projects that want to include it with that caveat.

Reply:
I was cautious to not depend on any CMake generated compile definition.
All macro's are in the form `SDL_DISABLE_xxx` or `SDL_ENABLE_ALTIVEC`.